src/Microsoft.Deployment.DotNet.Releases/src/ReleaseVersion.cs

```diff
@@ -693,144 +670,247 @@ internal static bool IsPrereleaseIdentifier(string input)
         /// <param name="value">The converted value.</param>
         /// <returns><see langword="true"/> if the string could be converted; <see langword="false"> otherwise.</see> </returns>
-        /// <exception cref="FormatException" />
-        /// <exception cref="ArgumentOutOfRangeException" />
-        internal static bool TryParseCoreVersionPart(string input, bool canThrow, out int value)
+        /// <exception cref="FormatException">If the input string is an invalid major, minor or patch value.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If <paramref name="throwOnFailure"/> is <see langword="true"/> and the value is less than 0.</exception>
+        /// <exception cref="OverflowException">If <paramref name="throwOnFailure"/> is <see langword="true"/> and the value exceeds <see cref="int.MaxValue"/>.</exception>
+        internal static bool TryParseCoreVersionPart(string input, bool throwOnFailure, out int value)
         {
             value = -1;
 
-            try
+            if (!IsNumericIdentifier(input))
             {
-                if (!IsNumericIdentifier(input))
+                if (throwOnFailure)
                 {
                     throw new FormatException(string.Format(ReleasesResources.InvalidNumericIdentifier, input));
                 }
 
+                return false;
+            }
+
+            if (throwOnFailure)
+            {
                 value = int.Parse(input);
+            }
+            else
+            {
+                if (!int.TryParse(input, out value))
+                {
+                    return false;
+                }
+            }
 
-                if (value < 0)
+            if (value < 0)
+            {
+                if (throwOnFailure)
                 {
                     throw new ArgumentOutOfRangeException(nameof(input), ReleasesResources.VersionPartLessThanZero);
                 }
+
+                return false;
             }
-            catch
+
+            return true;
+        }
+
+        private static bool IsValidPreleae(string prerelease)
+        {
+            if (prerelease != null)
             {
-                if (canThrow)
+                string[] dotSeparatedPrereleaseIdentifiers = prerelease.Split('.');
+
+                if (dotSeparatedPrereleaseIdentifiers.Length > 0)
                 {
-                    throw;
+                    for (int i = 0; i < dotSeparatedPrereleaseIdentifiers.Length; i++)
+                    {
+                        if (!IsPrereleaseIdentifier(dotSeparatedPrereleaseIdentifiers[i]))
+                        {
+                            return false;
+                        }
+                    }
                 }
+            }
 
-                return false;
+            return true;
+        }
+
+        private static bool IsValidBuildMetadata(string buildMetadata)
```

Comment:
Have you considered utilizing an expression bodied method?  It is a preference call.

```
private static bool IsValidBuildMetadata(string buildMetadata) => buildMetadata?.Split('.').All(part => IsBuildIdentifier(part)) ?? true;
```